ui/org.eclipse.pde.launching/src/org/eclipse/pde/internal/launching/JUnitLaunchValidationOperation.java

```diff
@@ -0,0 +1,112 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Simeon Andreev and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Simeon Andreev - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.pde.internal.launching;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.debug.core.DebugPlugin;
+import org.eclipse.debug.core.ILaunchConfiguration;
+import org.eclipse.debug.core.IStatusHandler;
+import org.eclipse.osgi.service.resolver.BundleDescription;
+import org.eclipse.osgi.util.NLS;
+import org.eclipse.pde.core.plugin.IPluginModelBase;
+import org.eclipse.pde.internal.launching.launcher.LaunchValidationOperation;
+
+public class JUnitLaunchValidationOperation extends LaunchValidationOperation {
+
+	private static final Set<String> JUNIT_JUPITER_BUNLDES = new HashSet<>(Arrays.asList(new String[] {"junit-jupiter-api", //$NON-NLS-1$
+			"org.junit.jupiter.api", //$NON-NLS-1$
+			"junit-jupiter-engine", //$NON-NLS-1$
+			"org.junit.jupiter.engine", //$NON-NLS-1$
+	}));
+
+	private static final Object[] EMPTY = new Object[0];
+	private final Map<Object, Object[]> fErrors = new HashMap<>(2);
+
+	public JUnitLaunchValidationOperation(ILaunchConfiguration configuration, Set<IPluginModelBase> models) {
+		super(configuration, models, null);
+	}
+
+	@Override
+	public void run(IProgressMonitor monitor) throws CoreException {
+		super.run(monitor);
```

Comment:
Is the validation performed in the super-class actually necessary for this?
It's at least called later in the launch procress again and we should avoid running the same validation twice for performance reasons.